static/app/views/settings/project/projectKeys/projectKeyCredentials.tsx

```diff
@@ -144,81 +139,320 @@ function ProjectKeyCredentials({
+      inline={false}
+      flexibleControlStateSize
+    >
+      <TextCopyInput aria-label={t('Unreal Engine Endpoint URL')}>
+        {unrealEndpoint}
+      </TextCopyInput>
+    </FieldGroup>
+  );
+}
+
+interface CredentialsTabProps {
+  projectId: string;
+  publicKey: string;
+  secretKey: string;
+  showProjectId: boolean;
+  showPublicKey: boolean;
+  showSecretKey: boolean;
+}
+
+function CredentialsTab({
+  publicKey,
+  secretKey,
+  projectId,
+  showPublicKey,
+  showSecretKey,
+  showProjectId,
+}: CredentialsTabProps) {
+  const fields = useMemo(() => {
+    return [
+      {
+        label: t('Public Key'),
+        value: publicKey,
+        show: showPublicKey,
+      },
+      {
+        label: t('Secret Key'),
+        value: secretKey,
+        show: showSecretKey,
+      },
+      {
+        label: t('Project ID'),
+        value: projectId,
+        show: showProjectId,
+      },
+    ];
+  }, [publicKey, secretKey, projectId, showPublicKey, showSecretKey, showProjectId]);
+  return (
+    <Fragment>
+      {fields.map(
+        field =>
+          field.show && (
+            <FieldGroup
+              key={field.label}
+              label={field.label}
+              inline={false}
+              flexibleControlStateSize
+            >
+              <TextCopyInput aria-label={field.label}>{field.value}</TextCopyInput>
+            </FieldGroup>
+          )
       )}
+    </Fragment>
+  );
+}
 
-      {showMinidump && (
+function ProjectKeyCredentials({
+  data,
+  projectId,
+  showDsn = true,
+  showDsnPublic = true,
+  showMinidump = true,
+  showProjectId = false,
+  showPublicKey = false,
+  showSecretKey = false,
+  showOtlpTraces = false,
+  showOtlpLogs = false,
+  showSecurityEndpoint = true,
+  showUnreal = true,
+}: Props) {
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  // Calculate available tabs based on props
+  const availableTabs = useMemo<TabConfig[]>(() => {
+    const tabs: TabConfig[] = [
+      {
+        key: 'credentials',
+        label: t('Credentials'),
+        visible: showPublicKey && showSecretKey && showProjectId,
+      },
+      {
+        key: 'otlp',
+        label: t('OpenTelemetry (OTLP)'),
+        visible: showOtlpTraces || showOtlpLogs,
+      },
+      {
+        key: 'security',
+        label: t('Security Header'),
+        visible: showSecurityEndpoint,
+      },
+      {
+        key: 'minidump',
+        label: t('Minidump'),
+        visible: showMinidump,
+      },
+      {
+        key: 'unreal',
+        label: t('Unreal Engine'),
+        visible: showUnreal,
+      },
+    ];
+    return tabs.filter(tab => tab.visible);
+  }, [
+    showOtlpTraces,
+    showOtlpLogs,
+    showSecurityEndpoint,
+    showMinidump,
+    showUnreal,
+    showPublicKey,
+    showSecretKey,
+    showProjectId,
+  ]);
+
+  // Get showDeprecatedDsn from query params
+  const showDeprecatedDsn = decodeScalar(location?.query?.showDeprecated) === 'true';
```

Comment:
note: it would’ve been cool if we started to use `nuqs` for this. It’s a new pattern, you can read about it in the developer docs here: https://develop.sentry.dev/frontend/url-state